pkg/options/options.go

```diff
@@ -57,8 +67,126 @@ type GCPAPIKey struct {
 }
 
 type AzureAPIKey struct {
-	BaseDnsDomain        string `yaml:"baseDnsDomain"`
-	BaseDomainRGN        string `yaml:"azureBaseDomainRGN"`
-	ServicePrincipalJson string `yaml:"azureServicePrincipalJson"`
-	Region               string `yaml:"region"`
+	BaseDnsDomain  string `yaml:"baseDnsDomain"`
+	BaseDomainRGN  string `yaml:"azureBaseDomainRGN"`
+	ClientId       string `yaml:"clientId"`
+	ClientSecret   string `yaml:"clientSecret"`
+	TenantId       string `yaml:"tenantId"`
+	SubscriptionId string `yaml:"subscriptionId"`
+	Region         string `yaml:"region"`
+}
+
+const charset = "abcdefghijklmnopqrstuvwxyz" +
+	"0123456789"
+
+var TestOptions TestOptionsT
+
+func LoadOptions(optionsFile string) error {
+	err := unmarshal(optionsFile)
+	if err != nil {
+		klog.Errorf("--options error: %v", err)
+		return err
+	}
+	return nil
+}
+
+func unmarshal(optionsFile string) error {
+
+	if optionsFile == "" {
+		optionsFile = os.Getenv("OPTIONS")
+		if optionsFile == "" {
+			optionsFile = "resources/options.yaml"
+		}
+	}
+
+	klog.V(2).Infof("options filename=%s", optionsFile)
+
+	data, err := ioutil.ReadFile(filepath.Clean(optionsFile))
+	if err != nil {
+		return err
+	}
+
+	err = yaml.Unmarshal([]byte(data), &TestOptions)
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func GetClusterName(cloud string) (string, error) {
+	var ownerPrefix string
+	// OwnerPrefix is used to help identify who owns deployed resources
+	//    If a value is not supplied, the default is OS environment variable $USER
+	if libgocmd.End2End.Owner != "" {
+		ownerPrefix = libgocmd.End2End.Owner
+	} else {
+		if TestOptions.ManagedClusters.Owner == "" {
+			ownerPrefix = os.Getenv("USER")
+			if ownerPrefix == "" {
+				ownerPrefix = "ginkgo"
+			}
+		} else {
+			ownerPrefix = TestOptions.ManagedClusters.Owner
+		}
```

Comment:
you can flip the logic here.. so one nested if will be reduced
```
if TestOptions.ManagedClusters.Owner != "" {
   ownerPrefix = TestOptions.ManagedClusters.Owner
}
ownerPrefix = os.Getenv("USER")
if ownerPrefix == "" {
   ownerPrefix = "ginkgo"
}
```

Reply:
with the above code as `ownerPrefix = os.Getenv("USER")` will always overwrite the value given in the TestOptions

Reply:
Maybe you meant:

`		ownerPrefix = os.Getenv("USER")
		if TestOptions.Owner != "" {
			ownerPrefix = TestOptions.Owner
		}
		if ownerPrefix == "" {
			ownerPrefix = "ginkgo"
		}
`

Reply:
okay.. I missed that part